### PR TITLE
pdm: 2.4.6 -> 2.5.2

### PR DESCRIPTION
**Pull request by betaboon**
###### Description of changes
https://github.com/pdm-project/pdm/releases/tag/2.5.0
https://github.com/pdm-project/pdm/releases/tag/2.5.1
https://github.com/pdm-project/pdm/releases/tag/2.5.2

supersedes #224324
 
cc @natsukium 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
The release notes should go into the commit message, as per https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md?rgh-link-date=2023-04-09T11%3A40%3A35Z.

**Comment by betaboon** [comment]
> The release notes should go into the commit message, as per https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md?rgh-link-date=2023-04-09T11%3A40%3A35Z.

done.

never knew about that requirement

**Comment by natsukium** [comment]
I don't know why you need another new PR just a few hours after the PDM release.
It seems to me that the `installer` should not be removed as it will be updated with `poetry` and might break the package again.

**Comment by betaboon** [comment]
@natsukium i didnt mean to offend anybody. i just had the time and thought that might help. 

regarding `installer`: i believe the potential for future breakage should not be the reason for having those individual dependencies.

**Comment by natsukium** [comment]
@betaboon 
I apologize for the aggressive tone.
It is important that the package is updated properly, so let's continue with this PR.

Since `pdm` is an application and `installer` is not yet a major release, I think it would be good to keep the explicit version pinning.
However, since the `installer` does not seem to be updated frequently, I feel that pinning is unnecessary now.

File: pkgs/tools/package-management/pdm/default.nix [hunk 1]
```diff
@@ -7,22 +7,13 @@ let
...
     packageOverrides = self: super: {
-      installer = super.installer.overridePythonAttrs (attrs: rec {
-        version = "0.6.0";
-        src = fetchFromGitHub {
-          owner = "pradyunsg";
-          repo = "installer";
-          rev = version;
-          hash = "sha256-IXznSrc/4LopgZDGFSC6cAOCbts+siKpdl5SvN1FFvA=";
-        };
-      });
```
Comment (mweinelt) [inline pkgs/tools/package-management/pdm/default.nix:18]: If you need 0.6.0, as opposed to 0.7.0, please keep this in. There will be another python updates run shortly, and we'll arrive at 0.7.0 for the release.

Comment (betaboon) [reply pkgs/tools/package-management/pdm/default.nix:18]: pdm uses `installer >= 0.7` now

Comment (mweinelt) [reply pkgs/tools/package-management/pdm/default.nix:18]: And we already have 0.7.0. Sorry for the noise.

**Pull request merged**